vm/src/stdlib/socket.rs

```diff
@@ -658,6 +658,32 @@ fn socket_getprotobyname(name: PyStringRef, vm: &VirtualMachine) -> PyResult {
     Ok(vm.ctx.new_int(num))
 }
 
+fn socket_getnameinfo(
+    address: Address,
+    flags: i32,
+    vm: &VirtualMachine,
+) -> PyResult<(String, String)> {
+    match get_addr(vm, address)?.as_std() {
+        Some(sock_addr) => match dns_lookup::getnameinfo(&sock_addr, flags) {
+            Ok((hostname, port)) => Ok((hostname, port)),
+            _ => {
+                let error_type = vm.class("_socket", "gaierror");
+                Err(vm.new_exception_msg(
+                    error_type,
+                    "nodename nor servname provided, or not known".to_owned(),
+                ))
+            }
+        },
+        _ => {
+            let error_type = vm.class("_socket", "gaierror");
+            Err(vm.new_exception_msg(
+                error_type,
+                "nodename nor servname provided, or not known".to_owned(),
+            ))
+        }
+    }
+}
```

Comment:
Because they share the errors, creating error can be moved out of match
```suggestion
    let addr = get_addr(vm, address)?;
    let nameinfo = addr.as_std().and_then(|addr| dns_lookup::getnameinfo(&addr, flags).ok());
    nameinfo.ok_or_else(|| {
          let error_type = vm.class("_socket", "gaierror");
          Err(vm.new_exception_msg(
              error_type,
              "nodename nor servname provided, or not known".to_owned(),
          ))
     })
```

Reply:
Thank you for your review. I revised it as you told me.